compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -336,7 +336,7 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 return true;
             }
         }
-        return false;
```

Comment:
IMO it is nicer, for symmetry with the `return` just a few lines above, to use `return` here.

That's exactly why I don't like just forcing such clippy lints over the entire codebase...

Reply:
I have exactly the opposite opinion. I like the consistency that tools like clippy encourage, and in this case I think the change makes the "early return" and "default/fallthrough" visually distinct, which I find easier to read.

Reply:
The last two returns here are not distinct in a meaningful sense, so it is not a good idea to make them visually distinct.

Reply:
I agree with @saethlin. I think it's just plain odd to have `return value;` at the end of a block when it's not suggesting exceptional control flow.

I think it's actually very odd that miri has `allow(clippy::needless_return)` on its codebase, but for the *rest* of the compiler I think this should be fixed.